File: src/types/application.ts
export interface Application {
    id: number;
    title: string;
    command: string[];
    img_square: string;
    categories: string[];
    playtime: number;

    last_launched?: Date;
    data_path?: string;

    steam_details?: SteamDetails;
    steam_drid_db_data?: SteamGridDbData;
    how_long_to_beat?: HowLongToBeat;
}

export interface SteamDetails {
    img_steam_horizontal?: string; // 92:43
    screeenshots?: string[];

    short_description?: string;
    detailed_description?: string;

    release_date?: Date;

    developers?: string[];
    publishers?: string[];

    pc_requirements?: {
        minimal: string;
        recommended: string;
    };
    linux_requirements?: {
        minimal: string;
        recommended: string;
    };
    controller_support?: string;

    achievements?: Achievement[];
}

export interface SteamGridDbData {
    img_hero?: string; // 16:9
}

export interface HowLongToBeat {
    main_story: number;
    main_plus_sides: number;
    completionist: number;
}

export interface Achievement {
    img: string;
    name: string;
    achieved: boolean;
}

File: src-tauri/src/models/application.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Achievement {
    pub img: String,
    pub name: String,
    pub achieved: bool,
}

#[derive(Debug, Serialize, Deserialize, Clone, Default)]
pub struct Requirements {
    pub minimal: String,
    pub recommended: String,
}

#[derive(Debug, Serialize, Deserialize, Clone, Default)]
pub struct HowLongToBeat {
    pub main_story: f64,
    pub main_plus_sides: f64,
    pub completionist: f64,
}

#[derive(Debug, Serialize, Deserialize, Clone, Default)]
pub struct SteamGridDbData {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub img_hero: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub img_steam_horizontal: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone, Default)]
pub struct SteamDetails {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub screeenshots: Option<Vec<String>>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub short_description: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub detailed_description: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub release_date: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub developers: Option<Vec<String>>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub publishers: Option<Vec<String>>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub pc_requirements: Option<Requirements>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub linux_requirements: Option<Requirements>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub achievements: Option<Vec<Achievement>>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Application {
    pub id: u64,
    pub title: String,
    pub command: Vec<String>,
    pub img_square: String,
    pub categories: Vec<String>,
    pub playtime: u64,

    // dayjs().toDate()
    #[serde(skip_serializing_if = "Option::is_none")]
    pub last_launched: Option<chrono::DateTime<chrono::Utc>>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub data_path: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub steam_details: Option<SteamDetails>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub how_long_to_beat: Option<HowLongToBeat>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub steam_drid_db_data: Option<SteamGridDbData>, 
}